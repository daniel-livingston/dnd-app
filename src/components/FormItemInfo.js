import React from "react";
import PropTypes from "prop-types";

const propTypes = {
	title: PropTypes.string.isRequired,
	description: PropTypes.string,
};

const FormItemInfo = (props) => (
	<div className='FormItemInfo'>
		<h3 className='FormItemInfo-Title'>{props.title}</h3>
		{props.description && <p className='FormItemInfo-Description'>{props.description}</p>}
	</div>
);

FormItemInfo.propTypes = propTypes;

export default FormItemInfo;
