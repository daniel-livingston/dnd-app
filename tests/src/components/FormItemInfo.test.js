import React from "react";
import { shallow } from "enzyme";
import FormItemInfo from "../../../src/components/FormItemInfo";

const props = {
	title: "Test title",
	description: "Test description",
};

describe("<FormItemInfo>", () => {
	test("renders correctly", () => {
		const wrapper = shallow(<FormItemInfo {...props} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("p")).toHaveLength(1);
	});

	test("renders correctly when no description provided", () => {
		const wrapper = shallow(<FormItemInfo title={props.title} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("p")).toHaveLength(0);
	});
});
