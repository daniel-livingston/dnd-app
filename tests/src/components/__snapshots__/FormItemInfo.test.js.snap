// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormItemInfo> renders correctly 1`] = `
<div
  className="FormItemInfo"
>
  <h3
    className="FormItemInfo-Title"
  >
    Test title
  </h3>
  <p
    className="FormItemInfo-Description"
  >
    Test description
  </p>
</div>
`;

exports[`<FormItemInfo> renders correctly when no description provided 1`] = `
<div
  className="FormItemInfo"
>
  <h3
    className="FormItemInfo-Title"
  >
    Test title
  </h3>
</div>
`;
